model/lifecycle.go

```diff
@@ -869,7 +909,7 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 	taskMap := make(map[string]*task.Task)
 	for _, t := range tasksToCreate {
 		id := execTable.GetId(b.BuildVariant, t.Name)
-		activateTask := len(tasksToActivate) == 0 || utility.StringSliceContains(tasksToActivate, t.Name)
+		activateTask := b.Activated && !utility.StringSliceContains(tasksWithBatchTime, t.Name)
```

Comment:
I think this also needs to handle the case where the task has a batchtime and it's elapsed right? Or is that handled elsewhere?

Reply:
If the batchtime has elapsed, that'll get hit by ActivateElapsedBuildsAndTasks.